app/models/project.rb

```diff
@@ -2,55 +2,118 @@
 class Project < ApplicationRecord
   validates_with ProjectValidator
   has_many :provenance_events, dependent: :destroy
+  before_save do |project|
+    # Ensure that the metadata JSONB postgres field is persisted properly
+    project.metadata = project.metadata_model
+  end
 
   # Valid project status described in ADR 7
   # See `architecture-decisions/0007-valid-project-statuses.md`
   PENDING_STATUS = "pending"
   APPROVED_STATUS = "approved"
   ACTIVE_STATUS = "active"
 
-  delegate :to_json, to: :metadata_json
+  delegate :to_json, to: :metadata_json # field in the database
+
+  def create!(initial_metadata:, user:)
+    self.metadata_model = initial_metadata
+    if initial_metadata.project_id.blank?
+        self.draft_doi(user: user)
```

Comment:
We should check to see if the project is valid before the DOI is drafted.  You are setting the DOI (which calls out to datacite) and then not saving that doi if the item is not valid.

Reply:
Fixed. We restored the logic to only mint the DOI once. 

We ended up declaring a constant `ProjectMetadata::DOI_NOT_MINTED` rather than using an empty string for the check.